src/extension/mapml/src/test/java/org/geoserver/mapml/MapMLWMSDataAccessProxyTest.java

```diff
@@ -0,0 +1,108 @@
+/* (c) 2024 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.mapml;
+
+import java.util.Collections;
+import java.util.List;
+import org.geoserver.catalog.Catalog;
+import org.geoserver.catalog.LayerInfo;
+import org.geoserver.catalog.WMSLayerInfo;
+import org.geoserver.catalog.WMSStoreInfo;
+import org.geoserver.data.test.SystemTestData;
+import org.geoserver.security.CatalogMode;
+import org.geoserver.security.DataAccessLimits;
+import org.geoserver.security.TestResourceAccessManager;
+import org.geotools.api.filter.Filter;
+import org.geotools.api.filter.FilterFactory;
+import org.geotools.factory.CommonFactoryFinder;
+import org.geotools.geometry.jts.JTSFactoryFinder;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.locationtech.jts.geom.GeometryFactory;
+import org.locationtech.jts.geom.Polygon;
+import org.locationtech.jts.io.WKTReader;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.User;
+
+public class MapMLWMSDataAccessProxyTest extends MapMLBaseProxyTest {
+
+    @BeforeClass
+    public static void beforeClass() {
+        initMockService(
+                "/mockgeoserver",
+                "/wms",
+                "REQUEST=GetCapabilities&VERSION=1.3.0&SERVICE=WMS",
+                "wmscaps.xml");
+    }
+
+    @Override
+    protected void setUpSpring(List<String> springContextLocations) {
+        super.setUpSpring(springContextLocations);
+        springContextLocations.add("classpath:/org/geoserver/wms/ResourceAccessManagerContext.xml");
+    }
+
+    @Override
+    protected void onSetUp(SystemTestData testData) throws Exception {
+        super.onSetUp(testData);
+        Catalog catalog = getCatalog();
+
+        WMSStoreInfo wmsStore = catalog.getFactory().createWebMapServer();
+        wmsStore.setName("wmsStore");
+        wmsStore.setWorkspace(catalog.getDefaultWorkspace());
+        wmsStore.setCapabilitiesURL(
+                "http://localhost:" + mockService.port() + getCapabilitiesURL());
```

Comment:
Every single usage of getCapabilitiesURL is a copy/paste of this concatenation. I'd recommend merging everything inside getCapabilitiesURL then.